tools/checkissuecomment/checkissuecomment.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main contains linter for todo issue comments.
+package main
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/singlechecker"
+)
+
+// todoIssueRegex matches potential TODO representations of an issue.
+var todoIssueRegex = regexp.MustCompile(`(?i)(?:issue[s]?|TODO|#):?\/?\s*\d+`)
+
+// validTodoIssueRegex matches correct TODO comment with the issue number.
+var validTodoIssueRegex = regexp.MustCompile(`^(.*?)\/\/\s*TODO https://github\.com/FerretDB/FerretDB/issues/\d+$`)
+
+var analyzer = &analysis.Analyzer{
+	Name: "checkissuecomment",
+	Doc:  "check for TODO comments with issue links",
+	Run:  run,
+}
+
+func main() {
+	singlechecker.Main(analyzer)
+}
+
+// run analyses the presence of TODO issue in code.
+func run(pass *analysis.Pass) (any, error) {
+	for _, file := range pass.Files {
+		fileName := pass.Fset.File(file.Pos()).Name()
+		f, err := os.Open(fileName)
+		if err != nil {
+			return nil, err
+		}
+
+		defer f.Close()
+
+		scanner := bufio.NewScanner(f)
+		lineNumber := 1
```

Comment:
Where is this variable used? I see it catches the number of the line that causes problems, but I don't see where we use the value to report which line causes a problem.

Reply:
@noisersup should I append line number in our print statement
`pass.Reportf(file.Pos(), "TODO comments mentioning issues not satisfied the pattern (// TODO <issue_url>): %s on line number: %d", strings.TrimSpace(line), lineNumber)`

Reply:
Yeah, otherwise we return the line 15 (the line of package declaration everywhere)

Reply:
I have added the line number

Reply:
Well, yeah, it is line 15 because we pass `file.Pos()`. It would be a correct line if we pass a correct line :)

Reply:
It turned out that the pass already contains all parsed comments; there is no need to read files manually. I went ahead and simplified the logic quite a bit.